girder/utility/acl_mixin.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright 2015 Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+import itertools
+import six
+
+from ..models.model_base import Model
+from ..constants import AccessType
+
+
+class AccessControlMixin(object):
+    """
+    This mixin is intended to be used for resources which aren't access
+    controlled by default, but resolve their access controls through other
+    resources. As such, the overridden methods retain the same parameters and
+    only alter functionality related to access control resolution.
+
+    resourceColl corresponds to the resource collection that needs to be used
+    for resolution, for example the Item model has a resourceColl of folder.
+
+    resourceParent corresponds to the field in which the parent resource
+    belongs, so for an item it would be the folderId.
+    """
+    resourceColl = None
+    resourceParent = None
+
+    def load(self, id, level=AccessType.ADMIN, user=None, objectId=True,
+             force=False, fields=None, exc=False):
+        """
+        Calls Model.load on the current item, and then attempts to load the
+        resourceParent which the user must have access to in order to load this
+        model.
+
+        Takes the same parameters as
+        :py:func:`girder.models.model_base.AccessControlledModel.load`.
+        """
+        doc = Model.load(self, id=id, objectId=objectId, fields=fields, exc=exc)
+
+        if doc is not None:
+            if self.resourceParent in doc and doc[self.resourceParent]:
+                loadType = self.resourceColl
+                loadId = doc[self.resourceParent]
+            else:
+                loadType = doc['attachedToType']
+                loadId = doc['attachedToId']
+
+        if not force and doc is not None:
+            self.model(loadType).load(loadId, level, user, objectId, force,
+                                      fields, exc)
```

Comment:
I addressed the excessive argument passing in afcc397 - but I'm a bit lost on what type of kwargs usage you're talking about.